src/main/webapp/app/modules/canvas/visualization/util/visualization-render-utils.ts

```diff
@@ -0,0 +1,207 @@
+import $ from 'jquery';
+import flairVisualizations from 'flair-visualizations/js/main';
+import Configuration from 'flair-visualizations/js/extras/configs/configuration';
+import { forwardCall } from 'app/shared/websocket/proxy-websocket.service';
+import { VisualWrap } from './visualmetadata-wrapper';
+const configuration = Configuration();
+
+export const ValidateFields = fields => {
+  let isValid = true;
+  fields.forEach(field => {
+    if (field.feature === null || undefined) {
+      isValid = false;
+      return isValid;
+    }
+  });
+  return isValid;
+};
+
+export const getVisualizationData = (visual, view) => {
+  if (visual.fields && ValidateFields(visual.fields)) {
+    const visualMetadata = VisualWrap(visual);
+    const queryDTO = visualMetadata.getQueryParameters(visual, null, null, null);
+    const body = {
+      queryDTO,
+      visualMetadata,
+      validationType: 'REQUIRED_FIELDS',
```

Comment:
keep this in constant file REQUIRED_FIELDS

Reply:
done